packages/components/src/components/Select/__tests__/Select.spec.tsx

```diff
@@ -0,0 +1,182 @@
+
+describe('useSelectData', () => {
+    const menuSlot = (
+        <Select.Slot key="test" name="menu">
+            <Select.Item value="test1">Test1</Select.Item>
+            <Select.Item value="test2">Test2</Select.Item>
+            <Select.Group groupId="Group 1">
+                <Select.Item value="test3">Test3</Select.Item>
+                <Select.Item value="test4">Test4</Select.Item>
+                <Select.Item value="test5">Test5</Select.Item>
+            </Select.Group>
+            <Select.Item value="test6">Test6</Select.Item>
+        </Select.Slot>
+    );
+
+    const decoratorsSlot = [
+        <Select.Slot key="test-left" name="left">
+            <IconIcon size={16} />
+        </Select.Slot>,
+        <Select.Slot key="test-right" name="right">
+            <IconIcon size={16} />
+        </Select.Slot>,
+    ];
+    const clearSlot = (
+        <Select.Slot key="test-clear" name="clear">
+            <IconIcon size={16} />
+        </Select.Slot>
+    );
+
+    const selectSlots = [menuSlot, clearSlot, ...decoratorsSlot];
+
+    it('returns menu slots', () => {
+        const {
+            result: { current: result },
+        } = renderHook(() => useSelectData(selectSlots));
+        expect(
+            Children.toArray(result.menuSlots).every((item) => {
+                return isValidElement(item) && (item.type === Select.Item || item.type === Select.Group);
+            }),
+        ).toBe(true);
+        expect(Children.toArray(result.menuSlots).length).toBe(4);
+        expect(result.items.length).toBe(6);
+    });
+
+    it('returns decorator slots', () => {
+        const {
+            result: { current: result },
+        } = renderHook(() => useSelectData(selectSlots));
+        expect(
+            result.inputSlots.every((decorator) => {
+                return isValidElement(decorator) && decorator.type === Select.Slot;
+            }),
+        ).toBe(true);
+        expect(result.inputSlots.length).toBe(2);
+    });
+
+    it('returns clear slot', () => {
+        const {
+            result: { current: result },
+        } = renderHook(() => useSelectData(selectSlots));
+        expect(isValidElement(result.clearButton) && result.clearButton.type === Select.Slot).toBe(true);
+    });
+
+    it('returns default item', () => {
+        const {
+            result: { current: result },
+        } = renderHook(() => useSelectData(selectSlots, 'test3'));
+        expect(result.defaultItem?.value).toBe('test3');
+    });
+
+    it('returns handles filterText', () => {
+        const { result } = renderHook(() => useSelectData(menuSlot));
+        expect(result.current.filterText).toBe('');
+        expect(result.current.items.length).toBe(6);
+        act(() => {
+            result.current.setFilterText('test1');
+        });
+        expect(result.current.filterText).toBe('test1');
+        expect(result.current.items.length).toBe(1);
+    });
+});
+
+describe('getSelectOptionValue', () => {
```

Comment:
Please move this `describe` to `utils.spec.ts`

Reply:
moved